### DISCUSSION TITLE
Integrating Gravity Bridge into Archway for Asset Transfers with Ethereum

### DISCUSSION THREAD

**Discussion by emperorjm** [archway-network/discussions#24] [General Discussions]
### Background:

Recently, the question was asked if it would be possible to integrate the Peggy bridge module into Archway, to streamline the transfer of Ethereum based assets in and out of Archway. It appears however that Peggy might have become obsolete, and Gravity Bridge seems to be the successor (https://github.com/cosmos/gravity-bridge?files=1). Gravity Bridge is designed to serve as a neutral Ethereum to Cosmos bridge and its unique proposition lies in its utilization of IBC enabling Cosmos projects to connect effortlessly.

### Rationale:

The rationale behind such an integration lies in the continuous drive to ensure efficient, secure, and user friendly transfers. As Ethereum remains one of the major players in the cryptocurrency market, facilitating smooth transactions between Archway and Ethereum can significantly enhance our utility and market presence.

### Questions:

- Considering the above context and potential advantages, the pressing question arises: "Is the integration of Gravity Bridge with Archway a technically and economically viable solution that aligns with our goals of security, efficiency, and enhanced user experience?”.
- What are the alternatives that should be considered?

This discussion aims to collectively explore the best path forward without making any preconceived assumptions. Your insights and experiences will be invaluable in shaping our strategy.

**Comment by ewais001** [comment]
Don't know enough about the costs, but advantages are significant. An integration would enable Astrovault to build out more pairs, enable more liquidity, and satisfy the needs of more customers on Archway.

**Comment by suhasagg** [comment]
Gravity Bridge is a trustless, neutral bridge between the Ethereum and Cosmos ecosystems. It is a purpose-built blockchain that uses the Cosmos SDK and IBC to bridge assets between the two chains. It also allows users to trade assets between Cosmos and Binance Chain.

To transfer assets from Ethereum to Cosmos using Gravity Bridge, users must first lock up their assets on the Ethereum side of the bridge. This creates a corresponding representation of those assets on the Cosmos side of the bridge. Users can then use these representations to send and receive assets on Cosmos.

To redeem their assets back to Ethereum, users must first burn their representations on the Cosmos side of the bridge. This releases the locked-up assets on the Ethereum side of the bridge.

**Mechanisms:**
**Batching**: Transactions are batched together before being sent to Ethereum, which reduces the cost and congestion on the Ethereum network.
**Liquidity pools**: Gravity Bridge maintains liquidity pools for each asset that is bridged. This ensures that users can always swap their representations for the underlying asset on either chain.

**How interchain accounts on Archway and Gravity Bridge can be beneficial:**

**Cross-chain gaming:** A game developer could create a cross-chain game that allows players to use their Gravity Bridge accounts to store and transfer in-game assets between Ethereum and Archway. This would allow players to seamlessly switch between playing the game on different chains.
**Cross-chain NFT marketplaces**: An NFT marketplace developer could create a cross-chain marketplace that allows users to trade NFTs between Ethereum and Archway. This would give users access to a wider pool of buyers and sellers.
**Cross-chain data sharing:** A data sharing developer could create an application that allows users to share data between Ethereum and Archway.

**Comment by suhasagg** [comment]
Evmos can act as a supplement and complement to Gravity Bridge in several ways:

Evmos is a fully EVM-compatible Cosmos chain. This means that developers can easily deploy Ethereum applications to Evmos without having to make any changes. This makes it a natural destination for Ethereum-based assets that want to move to Cosmos.

Evmos is a fully EVM-compatible Cosmos chain, which means that Ethereum applications can be deployed to Evmos without any changes. This makes Evmos a natural staging ground for Ethereum-based assets that want to move to Archway. Developers can test their applications on Evmos before migrating them to Archway, which can help to reduce the risk of errors.

EVMOS can be used as a testing ground for Binance-compatible dapps that want to move to Archway. Developers can test their applications on EVMOS before migrating them to Archway, which can help to reduce the risk of errors.

EVMOS can be used as a staging ground for Binance-compatible assets that want to move to Archway. Developers can gradually migrate their assets to EVMOS before migrating them to Archway, which can help to reduce the risk of disruptions.

**Reply by emperorjm** [reply]
@suhasagg, from your response, I see that Evmo's main utility is to allow Ethereum smart contracts to run within a chain that is connected to the Cosmos ecosystem and therefore would be able to interact via IBC for example.

I do have a few questions:

- Is it possible to bridge assets into Archway via Evmos?
- How does the user experience differ if one were to move assets via Evmos as opposed to a direct bridge like Gravity?
- Are there any associated potential challenges we should be aware of when considering Evmos as a staging ground?

**Comment by suhasagg** [comment]
@emperorjm
I am kindly adding my insights. 
Below is Evmo's bridges feature comparison sheet (It is a supplement and bridge will kindly be needed)  -
https://docs.google.com/spreadsheets/d/1CCha8tpKbvvSqERpiTKDaHnN4JzEhcIAd0gqKeoYFCM/edit?usp=sharing

Benefits of using Evmos as a staging ground  -

Interoperability: Evmos is a Cosmos chain, which means that it is interoperable with other Cosmos chains via IBC. This means that developers can build applications on Evmos that can interact with applications on other Cosmos chains.

Evmos is a fully EVM-compatible Cosmos chain, which means that developers can deploy their Ethereum applications to Evmos without having to make any changes. This can save developers a lot of time and effort.
